### DISCUSSION TITLE
customized exception body in openapi schema

### DISCUSSION THREAD

**Discussion by panla** [fastapi/fastapi#6763] [Questions] [answered]
Usual

```json
{
  "detail": [
    {
      "loc": [
        "string"
      ],
      "msg": "string",
      "type": "string"
    }
  ]
}
```

customized exception_handler

```python
@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request: Request, exc: RequestValidationError):

    exc_str = f'{exc}'.replace('\n', ' ').replace('   ', ' ')
    content = {'code': 10001, 'data': None, 'message': 'some message ...'}
    return JSONResponse(content=content, status_code=422)
```

the aim is 

```json
{
 "code": 10001,
 "data": ...,
 "message": ""
}
```
how to show like this in openapi schema?

**Comment by ycd** [comment, marked as answer]
see the [additional responses](https://fastapi.tiangolo.com/advanced/additional-responses/) section of the documentation.

**Comment by panla** [comment]
thanks, helpful

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! 👏  🙇 

Thanks for reporting back and closing the issue @panla  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
